### ISSUE TITLE
SyntaxError: Unexpected token

### ISSUE THREAD

**Issue by niftylettuce** [browserify/common-shakeify#18]
see the comment thread here for debug output

<https://github.com/visionmedia/superagent/issues/1418#issuecomment-424961250>

running this on `superagent`

**Comment by goto-bus-stop** [comment]
could you upload a sample project? between superagent, tinyify, babelify etc i'm not sure which things i need where in order to reproduce the issue.

**Comment by quinnturner** [comment]
I can confirm that a project (that I can't upload, sorry) with `babelify`, `tinyify` **running --no-flat**, and `superagent` has the same problem. Note that `tinyify` running without `--no-flat` works.

**Comment by josephg** [comment]
I've run into the same issue. This is a test case:

a.js:

```javascript
exports.x = a => a
exports.y = (a, b) => a ? b : !b
```

b.js:

```javascript
require('./a.js').x()
```

```
$ browserify -t uglifyify -p common-shakeify testcase/b.js > bundle.js
$ node bundle.js 
/Users/josephg/src/statecraft/bundle.js:2
exports.x=(x=>x),/* common-shake removed: exports.y = */ void 0, (x,e)=>x?e:!e);
                                                                              ^
SyntaxError: Unexpected token )
```

I can throw it all into a github repo if you need me to.

It looks like common-shakify is replacing this code:

```javascript
exports.y=((x,e)=>x?e:!e);
```

with this:

```javascript
/* common-shake removed: exports.y = */ void 0, (x,e)=>x?e:!e);
```

And the result ends up with a trailing `)`.

**Comment by goto-bus-stop** [comment]
Thanks, good work figuring that out! I prefer the PR you opened here to the change in common-shake.

**Comment by goto-bus-stop** [comment]
patched by @josephg in :package: 0.5.2. thanks all!

**Comment by josephg** [comment]
Awesome, thanks for merging! <3
